runtime/sema/type.go

```diff
@@ -1190,14 +1196,24 @@ type NumericType struct {
 	memberResolversOnce  sync.Once
 	saturatingArithmetic SaturatingArithmeticSupport
 	isSuperType          bool
+
+	// allow numeric types to conform to interfaces
+	conformances                         []*InterfaceType
+	effectiveInterfaceConformanceSet     *InterfaceSet
+	effectiveInterfaceConformanceSetOnce sync.Once
 }
 
 var _ Type = &NumericType{}
 var _ IntegerRangedType = &NumericType{}
 var _ SaturatingArithmeticType = &NumericType{}
 
 func NewNumericType(typeName string) *NumericType {
-	return &NumericType{name: typeName}
+	return &NumericType{
+		name: typeName,
+		conformances: []*InterfaceType{
+			StructStringerType,
+		},
+	}
```

Comment:
Maybe instead of creating a new, same conformance set for each numeric type, maybe consider not storing the conformance set (remove the new fields here), and return a global from `EffectiveInterfaceConformanceSet`, see below